pkg/cri/server/container_create_linux.go

```diff
@@ -166,10 +166,11 @@ func (c *criService) containerSpec(
 
 	// Apply envs from image config first, so that envs from container config
 	// can override them.
-	env := append([]string{}, imageConfig.Env...)
+	env := []string{}
 	for _, e := range config.GetEnvs() {
 		env = append(env, e.GetKey()+"="+e.GetValue())
 	}
+	specOpts = append(specOpts, oci.WithEnv(imageConfig.Env))
```

Comment:
I think we should use the copy of imageConfig.Env here. Since the imageConfig.Env is cached in memory, and it is hard to make sure that oci.WithEnv always uses readonly mode on the slice, we should use copy here in just case.

Reply:
Updated.